leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/MyDevice.java

```diff
@@ -90,12 +91,12 @@ public ReadResponse read(ServerIdentity identity, int resourceid) {
     }
 
     @Override
-    public ExecuteResponse execute(ServerIdentity identity, int resourceid, String params) {
-        String withParams = null;
-        if (params != null && params.length() != 0)
-            withParams = " with params " + params;
+    public ExecuteResponse execute(ServerIdentity identity, int resourceid, Arguments arguments) {
+        String withArguments = null;
+        if (!arguments.isEmpty())
+            withArguments = " with arguments " + arguments;
         LOG.info("Execute on Device resource /{}/{}/{} {}", getModel().id, getId(), resourceid,
-                withParams != null ? withParams : "");
+                withArguments != null ? withArguments : "");
```

Comment:
we should remove this test by initializing withArguments with ""

Reply:
You're right, It looks obvious now :smile: